ktor-network/posix/test/UdpSocketTest.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.tests
+
+import io.ktor.network.selector.*
+import io.ktor.network.sockets.*
+import io.ktor.util.network.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.*
+import kotlin.test.*
+
+class UdpSocketTest {
+    private val testJob = Job()
+    private val selector = SelectorManager(Dispatchers.Default + testJob)
+
+    @AfterTest
+    fun tearDown() {
+        testJob.cancel()
+        // TODO: This causes the [UdpSocketTest] to get stuck.
+        //selector.close()
+    }
+
+    @Test
+    fun test(): Unit = runBlocking {
+        withTimeout(15000) {
+            aSocket(selector).udp().bind(NetworkAddress("127.0.0.1", 8000)) {
+                reuseAddress = true
+                reusePort = true
+            }.use { socket ->
+                // Send messages to localhost
+                launch {
+                    val address = NetworkAddress("127.0.0.1", 8000)
+                    repeat(10) {
+                        val bytePacket = buildPacket { append("hello") }
+                        val data = Datagram(bytePacket, address)
+                        socket.send(data)
+                    }
+                }
+                // Receive messages from localhost
+                repeat(10) {
+                    val incoming = socket.receive()
+                    assertEquals("hello", incoming.packet.readText())
+                }
+            }
+        }
+    }
+
+    @Test
+    fun testBroadcast(): Unit = runBlocking {
+        withTimeout(15000) {
+            val serverSocket = CompletableDeferred<BoundDatagramSocket>()
+            val server = launch {
+                aSocket(selector)
+                    .udp()
+                    .bind(NetworkAddress("0.0.0.0", 56700))
+                    .use { socket ->
+                        serverSocket.complete(socket)
+                        val received = socket.receive()
+                        assertEquals("0123456789", received.packet.readText())
+                    }
+            }
+
+            serverSocket.await()
+
+            aSocket(selector)
+                .udp()
+                .bind(NetworkAddress("0.0.0.0", 0)) {
```

Comment:
Added this network address because of "Address auto-assign is not supported on Native. Consider specifying address manually.". Could that be added or is that not possible on native?